Backport of https://github.com/htop-dev/htop/pull/473

--- a/linux/LinuxProcessList.c	2021-01-11 20:43:28.186510057 +0100
+++ b/linux/LinuxProcessList.c	2021-01-14 13:48:23.452808449 +0100
@@ -1454,10 +1454,16 @@
       if (proc->state == 'Z' && (proc->basenameOffset == 0)) {
          proc->basenameOffset = -1;
          setCommand(proc, command, commLen);
+         lp->procCmdlineBasenameOffset = 0;
+         lp->procCmdlineBasenameEnd = 0;
+         lp->mergedCommand.commChanged = true;
       } else if (Process_isThread(proc)) {
          if (settings->showThreadNames || Process_isKernelThread(proc) || (proc->state == 'Z' && proc->basenameOffset == 0)) {
             proc->basenameOffset = -1;
             setCommand(proc, command, commLen);
+            lp->procCmdlineBasenameOffset = 0;
+            lp->procCmdlineBasenameEnd = 0;
+            lp->mergedCommand.commChanged = true;
          } else if (settings->showThreadNames) {
             if (! LinuxProcessList_readCmdlineFile(proc, procFd)) {
                goto errorReadingProcess;
